src/setup/init_fluid.f90

```diff
@@ -380,5 +381,49 @@ subroutine tuscan(lo, hi, vel, density, tracer, slo, shi, dx, dy, dz, domlo, dom
 
    end subroutine tuscan
 
+   subroutine channel_flow(lo, hi, vel, tracer, slo, shi, dx, dy, dz, domlo, domhi, xlo, ylo, zlo)
+
+      use constant, only: ic_u, ic_v, ic_w, ntrac,one
+
+      implicit none
+
+      integer(c_int),   intent(in   ) ::    lo(3),    hi(3)
+      integer(c_int),   intent(in   ) :: domlo(3), domhi(3)
+      integer(c_int),   intent(in   ) ::   slo(3),   shi(3)
+
+      real(rt),         intent(in   ) :: dx, dy, dz, xlo, ylo, zlo
+      real(rt),         intent(inout) ::    vel(slo(1):shi(1), slo(2):shi(2),slo(3):shi(3), 3)
+      real(rt),         intent(inout) :: tracer(slo(1):shi(1), slo(2):shi(2),slo(3):shi(3), ntrac)
+
+      ! Local variables
+      integer(c_int)                  :: i, j, k
+      integer(c_int)                  :: num_cells, num_cells_x, num_cells_y,num_cells_z
+      real(rt)                        :: x, y, z, yval, zval, pi
+
+      pi = 4.0_rt * atan(one)
+
+      do k = lo(3), hi(3)
+         z = zlo + dz*(real(k,rt) + 0.5d0) 
+         do j = lo(2), hi(2)
+            y = ylo + dy*(real(j,rt) + 0.5d0)
+            do i = lo(1), hi(1)
+               x = xlo + dx*(real(i,rt) + 0.5d0)
+
+               vel(i,j,k,1) = 8.0d0*(5.0d0*(1.0d0-z**4)/4.0d0 + 0.3d0*cos(5.0d0*pi*y)*exp(0.5-32.4*(1.0d0-abs(z))**2)*(1.0d0-abs(z)))
+               vel(i,j,k,2) = 8.0d0*21.6d0*sin(4.0d0*pi*x)*exp(-32.4d0*(1.0d0-abs(z))**2)*(1.0d0-abs(z))
+               vel(i,j,k,3) = 0.0d0
+               tracer(i,j,k,1:ntrac) = 0.0d0
```

Comment:
What are these for?

Reply:
the initial conditions for velocity that Mahesh used... similar to this page 9: https://www.mcs.anl.gov/~fischer/nek5000/sprague_nek5000_dec2010.pdf

Reply:
tracers would be for debugging tracers... Ann did a recent push that checks for ntrac>0

Reply:
@michaeljbrazell I was asking about the tracers, it wasn't clear what the values `1.0 - 3.0` stood for.

Reply:
ohh, sorry. I just stole that from plane_poiseuille. It has no meaning

Reply:
If we don't need the tracers, can we delete these lines?

Reply:
there will always be at least 1 tracer allocated... the flags come in through the input file to use more tracers incflo.ntrac and to advect the tracers incflo.advect_tracer. I was going to debug the tracer too so I'll leave just 1 tracer in there. If that's ok?